### ISSUE TITLE
Not OK: perl v5.6.0 on VMS_AXP V7.1

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#1655]
Migrated from [rt.perl.org#2895](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=2895) (status was 'resolved')

Searchable as RT2895$

**Comment by p5pRT** [comment]
### From @JohnPeacock



Overall a clean build under VMS \(WOW\!\!\!\)\.  If perlshr is already
defined \(i\.e\. an existing installation\)\, there are failures in
\[\.t\.lib\]vmsish\.t and \[\.t\.pragma\]warnings\.t \(Configure\.com should
check for this and suggest that perlshr be deassigned\)\.

Once that is done&#8203;: 

Failed two tests in \[\.t\.lib\]glob\-basic\.t&#8203;: 

Test 3 \- Patch below \(sorry for the lame VMS diff\)

File PERL\_SRC&#8203;:\[PERL\-5\_6\_0\.T\.LIB\]GLOB\-BASIC\.T;1
&nbsp;   42   if \($^O ne 'MSWin32' || $^O ne 'VMS'\) \{
&nbsp;   43     eval \{
\*\*\*\*\*\*
File PERL\_SRC&#8203;:\[PERL\-5\_6\_0\.T\.LIB\]GLOB\-BASIC\.T;2
&nbsp;   42   if \($^O eq 'MSWin32' || $^O eq 'VMS'\) \{
&nbsp;   43     print "ok 3 \# skipped\\n";
&nbsp;   44   \}
&nbsp;   45   else \{
&nbsp;   46     eval \{
\*\*\*\*\*\*\*\*\*\*\*\*
\*\*\*\*\*\*\*\*\*\*\*\*
File PERL\_SRC&#8203;:\[PERL\-5\_6\_0\.T\.LIB\]GLOB\-BASIC\.T;1
&nbsp;   52   \}
&nbsp;   53   print "ok 3\\n";
&nbsp;   54
\*\*\*\*\*\*
File PERL\_SRC&#8203;:\[PERL\-5\_6\_0\.T\.LIB\]GLOB\-BASIC\.T;2
&nbsp;   55     print "ok 3\\n";
&nbsp;   56   \}
&nbsp;   57
\*\*\*\*\*\*\*\*\*\*\*\*

Test 8 \- I do not understand what it is doing but @&#8203;a is 
being set to \('a'\,'a'\) so it fails all tests\.

Failed Tests 14\-16 in \[\.t\.lib\]vmsish\.t \(no patch supplied\)

not ok 14  \# \(time\) UTC&#8203;: 954450111  VMS&#8203;: 954432111
not ok 15  \# \(localtime\)
\# UTC&#8203;: 51 1 11 30 2 100 4 89 0
\# VMS&#8203;: 51 1 16 30 2 100 4 89 0
not ok 16  \# \(gmtime\)
\# UTC&#8203;: 51 1 16 30 2 100 4 89 0
\# VMS&#8203;: 51 1 21 30 2 100 4 89 0              

John Peacock

<details><summary>Perl Info</summary>

```

Flags:
    category=install
    severity=none

Site configuration information for perl v5.6.0:

Configured by jpeacock at Wed Mar 29 13:15:12 UTC-4:00 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
  Platform:
    osname=VMS, osvers=V7.1, archname=VMS_AXP
    uname='VMS kes V7.1 AlphaServer 4000 5/466 4MB'
    config_args=''
    hint=none, useposix=false, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef
usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=undef 
    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
  Compiler:
    cc='CC/DECC', optimize='', gccversion=undef
    cppflags='undef'
    ccflags ='/Include=[]/Standard=Relaxed_ANSI/Prefix=All/Obj=.obj/NoList'
    stdchar='char', d_stdstdio=define, usevfork=true
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=4, nvtype='double', nvsize=undef, Off_t='int',
lseeksize=4
    alignbytes=8, usemymalloc=N, prototype=define
  Linker and Libraries:
    ld='Link', ldflags ='/NoTrace/NoMap'
    libpth=/sys$share /sys$library
    libs= 
    libc=(DECCRTL), so=exe, useshrplib=undef, libperl=undef
  Dynamic Linking:
    dlsrc=dl_vms.c, dlext=exe, d_dlsymun=undef, ccdlflags=''
    cccdlflags='', lddlflags='/Share'

Locally applied patches:
    


@INC for perl v5.6.0:
    perl_root:[lib.VMS_AXP.5_6_0]
    perl_root:[lib]
    perl_root:[lib.site_perl.VMS_AXP]
    perl_root:[lib.site_perl]
    /perl_root/lib/site_perl
    .


Environment for perl v5.6.0:
    HOME=dkb0:[user.jpeacock]
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=perl_src:[perl-5_6_0.t.lib]
    PERLSHR=PERL_ROOT:[000000]PERLSHR.EXE
    PERL_BADLANG (unset)
    PERL_MODULES=DKB0:[USER.JPEACOCK.PERLMODULES.]
    PERL_ROOT=DRA0:[PERL5.]
    PERL_SRC=DKB0:[USER.JPEACOCK.PERLSRC.]
    SHELL (unset)



```
</details>

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]


In \<0003309544\.AA954455528@&#8203;univpress\.com> John Peacock wrote Thu\, 30 Mar 2000&#8203;:

> Overall a clean build under VMS \(WOW\!\!\!\)\.  If perlshr is already
> defined \(i\.e\. an existing installation\)\, there are failures in
> \[\.t\.lib\]vmsish\.t and \[\.t\.pragma\]warnings\.t \(Configure\.com should
> check for this and suggest that perlshr be deassigned\)\.

Some folks have reported clean builds and tests with a pre\-extant 
PERL\_ROOT and PERLSHR logical\.  In general the test suite is quite
careful about avoinding dependencies on PERL\_ROOT and PERLSHR to
avoid such upgrade problems\.

Is PERL\_ROOT=DRA0&#8203;:\[PERL5\.\] the new or old installation location?

> Once that is done&#8203;:
> 
> Failed two tests in \[\.t\.lib\]glob\-basic\.t&#8203;:
> 
> Test 3 \- Patch below \(sorry for the lame VMS diff\)
> 
> File PERL\_SRC&#8203;:\[PERL\-5\_6\_0\.T\.LIB\]GLOB\-BASIC\.T;1
>    42   if \($^O ne 'MSWin32' || $^O ne 'VMS'\) \{
>    43     eval \{
> \*\*\*\*\*\*
> File PERL\_SRC&#8203;:\[PERL\-5\_6\_0\.T\.LIB\]GLOB\-BASIC\.T;2
>    42   if \($^O eq 'MSWin32' || $^O eq 'VMS'\) \{
>    43     print "ok 3 \# skipped\\n";
>    44   \}
>    45   else \{
>    46     eval \{
> \*\*\*\*\*\*\*\*\*\*\*\*
> \*\*\*\*\*\*\*\*\*\*\*\*
> File PERL\_SRC&#8203;:\[PERL\-5\_6\_0\.T\.LIB\]GLOB\-BASIC\.T;1
>    52   \}
>    53   print "ok 3\\n";
>    54
> \*\*\*\*\*\*
> File PERL\_SRC&#8203;:\[PERL\-5\_6\_0\.T\.LIB\]GLOB\-BASIC\.T;2
>    55     print "ok 3\\n";
>    56   \}
>    57
> \*\*\*\*\*\*\*\*\*\*\*\*

Thanks \(did you test on Win32?\)\.

> Test 8 \- I do not understand what it is doing but @&#8203;a is
> being set to \('a'\,'a'\) so it fails all tests\.

For me on one Alpha test 7 fails when run under \`MMS test\`
but they all pass \(including 7\) when I type&#8203;:

&nbsp;   @&#8203;perl\_setup
&nbsp;   perl t/lib/vmsish\.t

> Failed Tests 14\-16 in \[\.t\.lib\]vmsish\.t \(no patch supplied\)
> 
> not ok 14  \# \(time\) UTC&#8203;: 954450111  VMS&#8203;: 954432111
> not ok 15  \# \(localtime\)
> \# UTC&#8203;: 51 1 11 30 2 100 4 89 0
> \# VMS&#8203;: 51 1 16 30 2 100 4 89 0
> not ok 16  \# \(gmtime\)
> \# UTC&#8203;: 51 1 16 30 2 100 4 89 0
> \# VMS&#8203;: 51 1 21 30 2 100 4 89 0

What does

&nbsp;  $ show logical sys$timezone\_differential

tell you about your system's UTC TDF?  I am in the PST so I have&#8203;:

&nbsp;  $ sho log sys$timezone\_differential
&nbsp;   "SYS$TIMEZONE\_DIFFERENTIAL" = "\-28800" \(LNM$SYSTEM\_TABLE\)

Peter Prymmer

**Comment by p5pRT** [comment]
### From @JohnPeacock

Peter \- 

I cannot say why I had problems with perlshr defined\, but as soon as I
deassign'd it\, most of the errors went away\.  The errors \(which I helpfully did
not save\) referred to problems activating the shared image PERLSHR\.EXE\.

I just tested the glob\-basic\.t patch on Win32 and it skipped the test \(as it
should\, apparently\)\.  I suspect that the test was originally for just one case
and someone hacked it quickly and blew the boolean logic\.  On any platform
except Win32\, the first half of the test is always true and the VMS half is
never tested because of the ||\.  I tend to prefer eq over ne because then the
sense of the || works better for me\.

I don't know grep well enough to understand what Test 8 is doing; I suspect it
is looking for a file called TES\* but why I only see two values @&#8203;a = \('a'\, 'b'\)\.
 Is there supposed to be a file called TEST\. in the directory someplace?  That
did not seem to help \(I created on in the t directory\)\.

My SYS$TIMEZONE\_DIFFERENTIAL was set to \-14400 for some unknown reason; that
appears to be the Atlantic Timezone\, not Eastern US\.  Can't find what script
that is in just yet\, but I'll track it down eventually\.  That fixed all of the
problems in VMSISH\.T\.  Thanks

John Peacock


\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_Reply Separator\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_
Subject&#8203;:    Re&#8203;: \[ID 20000330\.044\] Not OK&#8203;: perl v5\.6\.0 on VMS\_AXP V7\.1 
Author&#8203;: \<PVHP@&#8203;forte\.com>
Date&#8203;:       3/30/00 5&#8203;:30 PM


In \<0003309544\.AA954455528@&#8203;univpress\.com> John Peacock wrote Thu\, 30 Mar 2000&#8203;:

> Overall a clean build under VMS \(WOW\!\!\!\)\.  If perlshr is already
> defined \(i\.e\. an existing installation\)\, there are failures in
> \[\.t\.lib\]vmsish\.t and \[\.t\.pragma\]warnings\.t \(Configure\.com should
> check for this and suggest that perlshr be deassigned\)\.

Some folks have reported clean builds and tests with a pre\-extant 
PERL\_ROOT and PERLSHR logical\.  In general the test suite is quite
careful about avoinding dependencies on PERL\_ROOT and PERLSHR to
avoid such upgrade problems\.

Is PERL\_ROOT=DRA0&#8203;:\[PERL5\.\] the new or old installation location?

> Once that is done&#8203;:
> 
> Failed two tests in \[\.t\.lib\]glob\-basic\.t&#8203;:
> 
> Test 3 \- Patch below \(sorry for the lame VMS diff\)
> 
> File PERL\_SRC&#8203;:\[PERL\-5\_6\_0\.T\.LIB\]GLOB\-BASIC\.T;1
>    42   if \($^O ne 'MSWin32' || $^O ne 'VMS'\) \{
>    43     eval \{
> \*\*\*\*\*\*
> File PERL\_SRC&#8203;:\[PERL\-5\_6\_0\.T\.LIB\]GLOB\-BASIC\.T;2
>    42   if \($^O eq 'MSWin32' || $^O eq 'VMS'\) \{
>    43     print "ok 3 \# skipped\\n";
>    44   \}
>    45   else \{
>    46     eval \{
> \*\*\*\*\*\*\*\*\*\*\*\*
> \*\*\*\*\*\*\*\*\*\*\*\*
> File PERL\_SRC&#8203;:\[PERL\-5\_6\_0\.T\.LIB\]GLOB\-BASIC\.T;1
>    52   \}
>    53   print "ok 3\\n";
>    54
> \*\*\*\*\*\*
> File PERL\_SRC&#8203;:\[PERL\-5\_6\_0\.T\.LIB\]GLOB\-BASIC\.T;2
>    55     print "ok 3\\n";
>    56   \}
>    57
> \*\*\*\*\*\*\*\*\*\*\*\*

Thanks \(did you test on Win32?\)\.

> Test 8 \- I do not understand what it is doing but @&#8203;a is
> being set to \('a'\,'a'\) so it fails all tests\.

For me on one Alpha test 7 fails when run under \`MMS test\`
but they all pass \(including 7\) when I type&#8203;:

&nbsp;   @&#8203;perl\_setup
&nbsp;   perl t/lib/vmsish\.t

> Failed Tests 14\-16 in \[\.t\.lib\]vmsish\.t \(no patch supplied\)
> 
> not ok 14  \# \(time\) UTC&#8203;: 954450111  VMS&#8203;: 954432111
> not ok 15  \# \(localtime\)
> \# UTC&#8203;: 51 1 11 30 2 100 4 89 0
> \# VMS&#8203;: 51 1 16 30 2 100 4 89 0
> not ok 16  \# \(gmtime\)
> \# UTC&#8203;: 51 1 16 30 2 100 4 89 0
> \# VMS&#8203;: 51 1 21 30 2 100 4 89 0

What does

&nbsp;  $ show logical sys$timezone\_differential

tell you about your system's UTC TDF?  I am in the PST so I have&#8203;:

&nbsp;  $ sho log sys$timezone\_differential
&nbsp;   "SYS$TIMEZONE\_DIFFERENTIAL" = "\-28800" \(LNM$SYSTEM\_TABLE\)

Peter Prymmer

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

John Peacock wrote&#8203;:

> Peter \- 
> 
> I cannot say why I had problems with perlshr defined\, but as soon as I
> deassign'd it\, most of the errors went away\.  The errors \(which I helpfully did
> not save\) referred to problems activating the shared image PERLSHR\.EXE\.

That is a bit unfortunate\.  I ran \`mms test\` with my PERL\_ROOT and PERLSHR
pointing to a 5\.005\_02 installation\.

> I just tested the glob\-basic\.t patch on Win32 and it skipped the test \(as it
> should\, apparently\)\.  I suspect that the test was originally for just one case
> and someone hacked it quickly and blew the boolean logic\.  On any platform
> except Win32\, the first half of the test is always true and the VMS half is
> never tested because of the ||\.  I tend to prefer eq over ne because then the
> sense of the || works better for me\.

There was a problem with that test\.  My question was more of a rhetorical
one intended to remind you that testing ones proposed changes on all
affected platforms can be helpful \(it is not always necessary though\)\.

> I don't know grep well enough to understand what Test 8 is doing; I suspect it
> is looking for a file called TES\* but why I only see two values @&#8203;a = \('a'\, 'b'\)\.
>  Is there supposed to be a file called TEST\. in the directory someplace?  That
> did not seem to help \(I created on in the t directory\)\.

Take a look at Charles Lane's proposed fix for that test at&#8203;:

&nbsp;  http&#8203;://www\.xray\.mpe\.mpg\.de/mailing\-lists/vmsperl/2000\-03/msg00301\.html

you'll note that there is extra code to address test 8\.

> My SYS$TIMEZONE\_DIFFERENTIAL was set to \-14400 for some unknown reason; that
> appears to be the Atlantic Timezone\, not Eastern US\.  Can't find what script
> that is in just yet\, but I'll track it down eventually\.  That fixed all of the
> problems in VMSISH\.T\.  Thanks

Try looking for references to SYS$MANAGER&#8203;:UTC$CONFIGURE\_TDF\.COM\.
Note also that UTC does not go on and off DST so that this Sunday at 2&#8203;:00 AM
you ought to have a BATCH job set up to run that procedure for you \(being 
careful to slew the time by an hour \- the procedure will offer to do that 
for you\)\.  Good luck\.

Peter Prymmer
